cpp/src/arrow/python/pandas_convert.cc

```diff
@@ -1024,13 +1073,17 @@ class PandasBlock {
   Status AllocateNDArray(int npy_type, int ndim = 2) {
     PyAcquireGIL lock;
 
+    PyArray_Descr* descr = GetSafeNumPyDtype(npy_type);
+
     PyObject* block_arr;
     if (ndim == 2) {
       npy_intp block_dims[2] = {num_columns_, num_rows_};
-      block_arr = PyArray_SimpleNew(2, block_dims, npy_type);
+      block_arr = PyArray_NewFromDescr(
```

Comment:
There's also `PyArray_SimpleNewFromDescr(2, block_dims, descr)`, which looks a little less verbose to my eye.